File: app/src/main/java/cn/mycommons/androidmodular/InjectHelper.java
package cn.mycommons.androidmodular;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import cn.mycommons.androidmodular.app.AppContext;
import cn.mycommons.modulebase.IModuleConfig;

/**
 * InjectHelper <br/>
 * Created by xiaqiulei on 2017-05-14.
 */
public class InjectHelper {

    @NonNull
    public static AppContext getAppContext() {
        return AppContext.getAppContext();
    }

    @NonNull
    public static IModuleConfig getIModuleConfig() {
        return getAppContext().getModuleConfig();
    }

    @Nullable
    public static <T> T getInstance(Class<T> tClass) {
        IModuleConfig config = getIModuleConfig();
        Class<? extends T> implementClass = config.getServiceImplementClass(tClass);
        if (implementClass != null) {
            try {
                return implementClass.newInstance();
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        return null;
    }

    @Nullable
    public static <T> T getInstanceByDatabase(Class<T> tClass) {
        Implement_$$_Database database = new Implement_$$_Database();
        Class<? extends T> implementClass = database.getServiceImplementClass(tClass);
        if (implementClass != null) {
            try {
                return implementClass.newInstance();
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        return null;
    }
}

File: test-module/app/src/main/java/cn/mycommons/androidmodular/InjectHelper.kt
package cn.mycommons.androidmodular

import cn.mycommons.androidmodular.app.AppContext
import cn.mycommons.androidmodular.app.getAppContext
import cn.mycommons.modulebase.IModuleConfig

/**
 * InjectHelper <br></br>
 * Created by xiaqiulei on 2017-05-14.
 */
object InjectHelper {

    val appContext: AppContext get() = getAppContext()

    val iModuleConfig: IModuleConfig get() = appContext.getModuleConfig()

    fun <T> getInstance(tClass: Class<T>): T? {
        val config = iModuleConfig
        val implementClass = config.getServiceImplementClass(tClass)
        if (implementClass != null) {
            try {
                return implementClass.newInstance()
            } catch (e: Exception) {
                e.printStackTrace()
            }

        }
        return null
    }

//    fun <T> getInstanceByDatabase(tClass: Class<T>): T? {
//        val database = `Implement_$$_Database`()
//        val implementClass = database.getServiceImplementClass(tClass)
//        if (implementClass != null) {
//            try {
//                return implementClass.newInstance()
//            } catch (e: Exception) {
//                e.printStackTrace()
//            }
//
//        }
//        return null
//    }
}